visualization/app/codeCharta/ui/scenarioDropDown/scenarioDropDown.component.ts

```diff
@@ -3,34 +3,124 @@
 import "./scenarioDropDown.component.scss"
 import { ScenarioHelper, Scenario } from "../../util/scenarioHelper"
 import { MetricService, MetricServiceSubscriber } from "../../state/metric.service"
-import { MetricData } from "../../codeCharta.model"
+import { ColorRange, MetricData } from "../../codeCharta.model"
 import { IRootScopeService } from "angular"
-import { ThreeOrbitControlsService } from "../codeMap/threeViewer/threeOrbitControlsService"
 import { StoreService } from "../../state/store.service"
 import { setState } from "../../state/store/state.actions"
+import { DialogService } from "../dialog/dialog.service"
+import { setColorRange } from "../../state/store/dynamicSettings/colorRange/colorRange.actions"
+import { ThreeOrbitControlsService } from "../codeMap/threeViewer/threeOrbitControlsService"
 
 export class ScenarioDropDownController implements MetricServiceSubscriber {
 	private _viewModel: {
 		scenarios: Scenario[]
+		isScenarioChanged: boolean
 	} = {
-		scenarios: null
+		scenarios: null,
+		isScenarioChanged: true //TODO: When Method implemented to detect that codemap has changed set it to false at first
 	}
 
+	private availableMetrics: MetricData[]
+
 	constructor(
 		private $rootScope: IRootScopeService,
 		private storeService: StoreService,
+		private dialogService: DialogService,
 		private threeOrbitControlsService: ThreeOrbitControlsService
 	) {
 		MetricService.subscribe(this.$rootScope, this)
 	}
 
+	public loadScenarios() {
+		this._viewModel.scenarios = ScenarioHelper.getScenarios()
+	}
+
 	public onMetricDataAdded(metricData: MetricData[]) {
-		this._viewModel.scenarios = ScenarioHelper.getScenarios(metricData)
+		this._viewModel.scenarios = ScenarioHelper.getScenarios()
+		this.availableMetrics = metricData
+	}
+
+	public getButtonColor() {
+		return this._viewModel.isScenarioChanged ? "black" : "gray"
```

Comment:
create a CSS class to have the point by default in the one color and with the class `is-scenario-changed` it gets the other color. And then use a boolean inside the viewModel. This way we can separate the styling and JS.